spring-cloud-starter-kubernetes-client/pom.xml

```diff
@@ -28,19 +12,10 @@
 	<artifactId>spring-cloud-starter-kubernetes-client</artifactId>
 
 	<dependencies>
-		<dependency>
```

Comment:
this will be more of a question here... `spring-cloud-kubernetes-commons` and `spring-cloud-kubernetes-client-autoconfig` are already transitive dependencies of `spring-cloud-kubernetes-client-discovery`. So should we still provide them? Seems like un-necessary?

Reply:
@ryanjbaxter would you be able to take a look here and provide some feedback please? thank you

Reply:
Seems to make sense but I would just make sure by creating a simple project and make sure we aren't missing any dependencies.

Reply:
hmm, but why? these starters are supposed to expose only the `spring-cloud-kubernetes-fabric8-discovery` - discovery functionality that is and all else must be an implementation detail. 

Unless you have something different in mind and I don't get the full picture :)